src/Presentation/Presenters/DonorUpdateHtmlPresenter.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Presentation\Presenters;
+
+use WMDE\Fundraising\DonationContext\Domain\Model\Donation;
+use WMDE\Fundraising\DonationContext\UseCases\UpdateDonor\UpdateDonorResponse;
+use WMDE\Fundraising\Frontend\Infrastructure\UrlGenerator;
+use WMDE\Fundraising\Frontend\Presentation\DonorDataFormatter;
+use WMDE\Fundraising\Frontend\Presentation\DonationMembershipApplicationAdapter;
+use WMDE\Fundraising\Frontend\Presentation\TwigTemplate;
+
+/**
+ * Render the confirmation pages for donations with additional form to enter / update donation data
+ *
+ * @license GNU GPL v2+
+ */
+class DonorUpdateHtmlPresenter {
+
+	private $template;
+	private $donationMembershipApplicationAdapter;
+	private $urlGenerator;
+	private $donorDataFormatter;
+
+	public function __construct( TwigTemplate $template, UrlGenerator $urlGenerator ) {
+		$this->template = $template;
+		$this->urlGenerator = $urlGenerator;
+		$this->donationMembershipApplicationAdapter = new DonationMembershipApplicationAdapter();
+		$this->donorDataFormatter = new DonorDataFormatter();
+	}
+
+	public function present( UpdateDonorResponse $updateDonorResponse, Donation $donation, string $updateToken, string $accessToken ): string {
+		return $this->template->render(
+			$this->getConfirmationPageArguments( $updateDonorResponse, $donation, $updateToken, $accessToken )
```

Comment:
Could you split the variables from `getConfirmationPageArguments` that are not part of the original confirmation page presenter into a separate function and use `array_merge` here to make it more clear what belongs where?